contracts/src/token/erc721/mod.rs

```diff
@@ -549,6 +555,13 @@ impl IErc721 for Erc721 {
     }
 }
 
+impl IErc165 for Erc721 {
```

Comment:
I am not sure if we want to have it implemented always.

Reply:
It is [implemented](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/530179a71f435e85ae9df9b9f12b5637cf229e5c/contracts/token/ERC721/ERC721.sol#L19) in solidity for Erc721

Reply:
We will have to decide on an approach for all the standards in our library, so we can use this thread to do so(I've also opened an internal chat to get our Solidity team opinion). I'm leaning more on not adding it by default and letting the user opt in when they want. 
That being said when the standard defines that `ERC165` should be part of the contract we should do it. That's the case with [ERC721](https://eips.ethereum.org/EIPS/eip-721)

Reply:
@qalisander after discussing internally and doing some research it seems `ERC165` is not as widely used that is worth adding it to all our contract. What I suggest we do:
1. Add it automatically to **only** the standards that require it as part of the ERC(e.g ERC721)
2. Add a section in our docs that explains how to add it to your own contract. This could be under the API reference for ERC165 or similar.

What do you think?

Reply:
I think that would look a bit inconsistent. I even wondering do we need `ERC165` that much just for `ERC721` .. Also I suspect, it is not widely used mostly, because it is not implemented for the contracts like erc20.
1.  Consider that rust is different from solidity. And with current implementation just to make `support_interface` function visible, library user should import `openzeppelin_stylus::utils::introspection::erc165::IErc165` module in the scope (otherwise it won't be even compiled). And then reexport it within `#[public]` macro. So we are not "automatically" implementing the standard as I see, but making it simple to implement for users.
2. That's a great point. I need to add a guide about erc165 to antora docs. But seems for me the `ERC165` guide (for erc20) would be more simple if we implement `IErc165` trait for `ERC20` also.

Reply:
And one more point. There is an "orphan rule" in rust that disallows to user implement foreign trait (like 'IErc165` in our case) for a foreign contract type (like `Erc20`). In this case the `IErc165` trait seems even less practical for users if they would try to implement it for Erc20. May be it would even make sense to remove it..

Reply:
> I think that would look a bit inconsistent. 
I don't think it is inconsistent if we explicitly say: We will only add it to standards that require it and give the user the option to add it to any contract(s) they think is worth it.

> I even wondering do we need ERC165 that much just for ERC721
It wouldn't be for ERC721 only, since the implementation would be available for anyone wanting to add this to contracts they would like to specify which interfaces they support.

> And then reexport it within #[public] macro. So we are not "automatically" implementing the standard as I see, but making it simple to implement for users.
What do you mean here? We are adding it to the #[public] impl block of contracts. That means it would automatically be part of the inherited contract api right?

Reply:
> What do you mean here? We are adding it to the #[public] impl block of contracts. That means it would automatically be part of the inherited contract api right?
We are doing so inside examples. User also can opt out to export it to the abi, even if trait `IErc165` is implemented in our library. That is the difference from solidity, where as I see `Erc721` will always have `supportInterface(...)` abi public

Reply:
Ok... after discussing this today I agree it makes sense to add it to implementations of main interfaces and it doesn't come at any cost to users that don't want it(since it is not part of the public abi and they still need to explicitely opt in). Only potential downside is a bit of gas overhead when inheriting from multiple contracts.